src/sentry/models/release.py

```diff
@@ -92,8 +92,56 @@ def _on_post(project, trigger):
         ):
             schedule_invalidate_project_config(project_id=project.id, trigger=trigger)
 
+    def _subscribe_project_to_alert_rule(project, release, trigger):
+        """
+        TODO: potentially enable custom query_extra to be passed on ReleaseProject creation (on release/deploy)
+        """
+        from django.utils import timezone
+
+        from sentry.incidents.logic import subscribe_projects_to_alert_rule
+        from sentry.incidents.models import (
+            AlertRule,
+            AlertRuleActivationConditionType,
+            AlertRuleMonitorType,
+        )
+
+        try:
+            project_alert_rules = AlertRule.filter(project=project)
+            if project_alert_rules.exists():
+                # If we have an AlertRule for the project
+                for par in project_alert_rules:
+                    if (
+                        par.monitor_type == AlertRuleMonitorType.ACTIVATED
+                        and par.activation_conditions
+                        == AlertRuleActivationConditionType.RELEASE_CREATION
```

Comment:
could we add these to the query itself?

Is the `activation_conditions` something like "delay for 30min"? I know one use case we need to keep in mind is being able to delay starting the query window until conditions are met, curious how that kind of delay would work in this design.

If we need to do more for those activation conditions, it would be nice to have the `subscribe_projects_to_alert_rule` function handle those cases internally.

Reply:
Yea - I would categorize those use cases as "advanced" activation conditions.
for now, i want to just get out the base case, but i left a TODO at the top of the method to start thinking about how we can include custom queries as well

Reply:
We should probably also think about separating the activation conditions from the query itself.

We probaly _dont_ want to include those advanced conditions in the query themselves since we shouldn't even have _created_ the query until the conditions have been resolved - otherwise we'd be creating a lot of unnecessary work for snuba